variables.tf

```diff
@@ -293,3 +293,9 @@ variable "attach_worker_cni_policy" {
   type        = bool
   default     = true
 }
+
+variable "worker_group_managed_node_groups" {
+  description = "A list of maps defining worker group configurations to be defined using AWS EKS Managed Node Groups. See workers_group_defaults for valid keys."
+  type        = any
```

Comment:
The type should probably be either [list(any)](https://www.terraform.io/docs/configuration/types.html#list-) or, better yet, [list(object({for_examples = follow_this_link}))](https://www.terraform.io/docs/configuration/variables.html#declaring-an-input-variable).

Reply:
I would make it list(any) but none of the other 2 config lists (`worker_groups` or `worker_groups_launch_template`) are listed that way. I usually think that its better for the code to fit in vs. be more technically correct. That way its a cohesive whole

Reply:
If I am going to change that, I should change the other 2 as well. Would that fall under this PR or be a different one?

Reply:
Do a separate commit. Altering the existing resources will be a breaking change for users and therefore require a major version increment, whereas introducing it here is 100% safe given that it's brand new functionality.

Reply:
I resolved this for managed worker groups but didn't touch this for any others

Reply:
Hi. This really should *not* be `list(any)`. `any` is a stupid name for what this type does. It really means "any of the same type" and Terraform will try to convert items to the same type. With `list(any)` it will try to convert to a single type, usually `string` if one is present. Or throw errors when incompatible types are used, say only booleans and numbers. This list can contain incompatible mixed types like sublists and numbers. When used as a bareword like originally, terraform will convert to an Object using the types given. 

`list(object())` is even worse. When you specify an `object` the caller has to exactly match the keys listed. It would be a massive pain for all users of this module.

`type = any` is the only correct answer here.

Reply:
I stand corrected. `type = any` it is then.